packages/compiler/src/server/update-manager.ts

```diff
@@ -58,11 +64,30 @@ export class UpdateManager<T = void> {
         return await this.#update(Array.from(updates.values()), arg);
       },
       () => (this.#isStarted ? "ready" : "pending"),
-      this.getAdaptiveDebounceDelay,
+      this.getDebounceDelayFromEnv() === -1
+        ? this.getAdaptiveDebounceDelay
```

Comment:
please also add a test to verify getAdaptiveDebounceDelay is used when no env var is set.